benches/arrays/sum.ncl

```diff
@@ -0,0 +1,12 @@
+let sum
+  | Array Num -> Num
+  = let rec aux = fun step xs =>
```

Comment:
If the goal of this bench is a case where a lot of array contracts are applied at each recursive call, I would not care about stack space/tail calls and write the naive sum function `sum [] = 0, sum x:xs = 1 + sum xs`. Here, the `Array Num` input contract is only applied on the call to `sum`, but not to actual recursive calls to `aux` (although there's still `array.length` which applies an array contract each times), I think.